Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/fragments/EmulationFragment.java

```diff
@@ -389,7 +392,22 @@ public synchronized void clearSurface()
           NativeLibrary.WaitUntilDoneBooting();
         }
 
-        NativeLibrary.SurfaceDestroyed();
+        // Handle AlertMessages while booting the core.
+        if (NativeLibrary.IsBooting())
+        {
+          if (NativeLibrary.GetAlertMessageText() != null)
+          {
+            throw new IllegalStateException(NativeLibrary.GetAlertMessageText());
```

Comment:
Yes, I'm not sure the best way to handle this. The choices I thought of were deadlocking for an unknown reason or crashing for a known reason. I also assume panic alerts during core booting indicate major emulation problems anyway. This does fix `AlertMessages` outside of core booting so I think it's a worthwhile change that isn't worse than what we currently have. I'm open to other suggestions though.

Reply:
I guess the other option would be to dismiss the dialog if you try to rotate the screen while showing an alert message while booting. While it does avoid crashing, it is a bit odd, so I'm not sure if it's actually better...

I guess either a crash or what I mentioned would be okay, but you should leave a comment about it in either case (either about this crash being intentional, or about why we dismiss the dialog).

Reply:
Also, if you're going to have it crash, it probably makes more sense to always include "Attempted to destroy surface while booting the core" in the exception message (perhaps with the alert message text appended), since the error that causes the alert message to be shown is not in itself the reason why we are crashing.

Reply:
I tried another idea. Instead of deadlocking or crashing the emulator, we can simply end emulation. The AlertMessage has been replaced with a log error and toast. Hopefully this won't cause unforeseen problems.

Reply:
I also moved the `R.string.emulation_touch_button_help` toast to only run once during emulation. This will prevent it from reappearing when other toasts are displayed.

Reply:
Ending emulation is also kinda weird, but I guess it's no worse than crashing.

What about just turning the alert message into a toast and then calling `WaitUntilDoneBooting` as usual, without stopping emulation or anything?

Reply:
If `WaitUntilDoneBooting` is called instead of stopping emulation, we'll get the deadlock. I think that is the least appealing option of deadlock vs. crash vs. ending emulation. Still debating if I should go back to crashing the emulatior as I'm also not certain of the safety of stopping emulation either.

Reply:
> If `WaitUntilDoneBooting` is called instead of stopping emulation, we'll get the deadlock.

Yes, unless we release the lock first. My suggestion is that we dismiss the dialog and release the lock, and then show a toast.

Reply:
I just realized that regardless of what we do about this case, we're still going to get a deadlock in the case where we get an alert message after `WaitUntilDoneBooting` already has been called... I wonder if we maybe just should show all alert messages that are triggered when booting as toasts without even attempting to use a proper dialog.

Reply:
I could use the new `NativeLibrary.IsBooting` check to handle alert messages differently during core boot. Just to clarify, this PR doesn't deadlock with alert messages after core boot because it skips calling `WaitUntilDoneBooting` for `AlertMessages`.

Reply:
Yes, there is no problem after we're done booting.

Reply:
Moving the implementation of `IsBooting` and making alert messages toasts while the core is booting seems to be a good solution here. Hopefully this version does the trick!